bundles/org.eclipse.osgi.tests/src/org/eclipse/osgi/tests/bundles/EquinoxBundleAdaptTests.java

```diff
@@ -42,17 +43,23 @@ public void testAdapt_Module() throws Exception {
 	public void testAdapt_ProtectionDomain() throws Exception {
 		Bundle bundle = installer.installBundle("test");
 		SecurityManager previousSM = System.getSecurityManager();
+		boolean unsetSecurityManager = false;
 		try {
-			System.setSecurityManager(new SecurityManager() {
-				@Override
-				public void checkPermission(Permission perm) {
-					// Just permit everything
-				}
-			});
+			if (SecurityManagerTests.isSecurityManagerAllowed()) {
```

Comment:
can we probabbly use something like `asumeThat(SecurityManagerTests.isSecurityManagerAllowed())` instead?

Reply:
For this test-case I wasn't 100% sure if it's useless if the security-manager is disallowed or removed.
It could still test that `bundle.adapt(java.security.ProtectionDomain.class)` returns not null, even when the  `SecurityManager` is still gone/disabled. But of course the main value is checking if the `SecurityManager` is queried for the corresponding permission.

Reply:
OK, it looks like this test does not work at all if a security-manager is not allowed, because a ProtectionDomain is never returned if the security-manager is null:
https://github.com/eclipse-equinox/equinox/blob/372338078b3d5c908fcc5ed39a10683e9ea10dff/bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/BundleInfo.java#L332-L335

Even better, just skipping the test is simpler.